libminifi/test/unit/OsUtilTests.cpp

```diff
@@ -0,0 +1,41 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#include "utils/OsUtils.h"
+#include "../TestBase.h"
+#include "../Catch.h"
+
+namespace org::apache::nifi::minifi::test {
+
+#ifdef WIN32
+TEST_CASE("Test userIdToUsername for well-known SIDs", "[OsUtils]") {
+  CHECK("Nobody" == minifi::utils::OsUtils::userIdToUsername("S-1-0-0"));
+  CHECK("Everyone" == minifi::utils::OsUtils::userIdToUsername("S-1-1-0"));
+  CHECK("Local" == minifi::utils::OsUtils::userIdToUsername("S-1-2-0"));
+  CHECK("Console Logon" == minifi::utils::OsUtils::userIdToUsername("S-1-2-1"));
+  CHECK("Creator Owner" == minifi::utils::OsUtils::userIdToUsername("S-1-3-0"));
+  CHECK("Creator Group" == minifi::utils::OsUtils::userIdToUsername("S-1-3-1"));
+  CHECK("CREATOR OWNER SERVER" == minifi::utils::OsUtils::userIdToUsername("S-1-3-2"));
+  CHECK("CREATOR GROUP SERVER" == minifi::utils::OsUtils::userIdToUsername("S-1-3-3"));
+  CHECK("OWNER RIGHTS" == minifi::utils::OsUtils::userIdToUsername("S-1-3-4"));
+  CHECK("NT SERVICE\\ALL SERVICES" == minifi::utils::OsUtils::userIdToUsername("S-1-5-80-0"));
```

Comment:
I believe these usernames are localized, so non-english installations will have different names there.

Reply:
`userIdToUsername` first checks a fixed set of identifiers in `resolve_common_identifiers`, which returns english users, so if localization becomes an issue, this test will fail and we will have to look into `resolve_common_identifiers` as well, created a ticket to handle this: [MINIFICPP-2013](https://issues.apache.org/jira/browse/MINIFICPP-2013)

Reply:
I managed to test this on a hungarian windows installation. It fails: https://pastebin.com/qHW8B7Et
I prefer not to check in a failing test.

Reply:
I like the new version with the emptiness checks. We can possibly change them to checking actual values in MINIFICPP-2013.